engines/macventure/macventure.cpp

```diff
@@ -958,9 +958,18 @@ Common::String MacVentureEngine::getCommandsPausedString() const {
 
 Common::Path MacVentureEngine::getFilePath(FilePathID id) const {
 	if (id <= 3) { // We don't want a file in the subdirectory
+
+		// Convert from MacRoman encoding to unicode for punycode encoding
+		Common::String result = Common::convertToU32String((_filenames->getString(id)).c_str(), Common::kMacRoman);
+
 		return Common::Path(_filenames->getString(id));
 	} else { // We want a game file
-		return Common::Path(_filenames->getString(3) + "/" + _filenames->getString(id));
+
+		// Convert from MacRoman encoding to unicode for punycode encoding
+		Common::String result = _filenames->getString(3) + "/" + _filenames->getString(id);
```

Comment:
Here, instead of doing this string composition, you call `append()` from Common::Path